pkg/dev/podmandev/podmandev.go

```diff
@@ -167,5 +168,8 @@ func (o *DevClient) watchHandler(ctx context.Context, pushParams adapters.PushPa
 		WatchFiles:   watchParams.WatchFiles,
 		Variables:    watchParams.Variables,
 	}
+
+	fmt.Fprintln(watchParams.Out)
+	log.Fwarning(watchParams.Out, "Changes to the Devfile not supported yet on Podman. Please restart 'odo dev' for such changes to be applied.\n")
```

Comment:
This will be displayed every time, even if the user only modifies sources files.

I would instead add an `else` part to the `if equality.Semantic.DeepEqual(o.deployedPod, pod)` condition (in the reconcile.go file), and print the warning there

Reply:
Ah yes, thanks for the pointer. Initially, I had changed the part that looks for changes in the Devfile watcher (in watch.go), but then I had an issue when manually sync'ing with `--no-watch`.
This looks like a better approach. I'll make the change - thanks.

Reply:
One thing I just noticed: if we display the warning as part of `if !equality.Semantic.DeepEqual(o.deployedPod, pod)`, the problem is that this would work only if the changes in the Devfile affect the resulting Pod spec.

I noticed that if I change for example the `run` command only in the Devfile, the resulting Pod spec is still semantically the same as the previous one, so no warning is displayed. But, the changes were not taken into account (I think because we are still using the same Devfile Object from the context).

Reply:
I am thinking of just re-parsing the Devfile when the watch handler is called and comparing it to the currently loaded one. And display the warning if there are any differences.
@feloy What do you think?

The Kubedev implementation actually re-parses the Devfile when the watch handler is called: 
https://github.com/redhat-developer/odo/blob/020009b33cbef46c143075f8d830bb0284f872c6/pkg/dev/kubedev/kubedev.go#L156-L160